### PR TITLE
Add package "graceful"

### PR DESCRIPTION
**Pull request by mraerino**
Adds a handler to shutdown daemon routines gracefully

This is mostly made for http.Server which exposes a Shutdown method taking a context for timeout

### CODE REVIEW

**Comment by rybit** [comment]
@mraerino do you still wanna take this in after we brought the router/server interfaces in?

**Comment by mraerino** [comment]
this PR reduces this part of the boilerplate pretty cleanly. i think it's useful.
https://github.com/netlify/go-service-boilerplate/blob/master/cmd/root.go#L55-L66

File: graceful/graceful.go [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+		signals := make(chan os.Signal, 1)
+		signal.Notify(signals, os.Interrupt, syscall.SIGTERM, syscall.SIGINT)
+		select {
+		case sig := <-signals:
+			log.Infof("Triggering shutdown from signal %s", sig)
+		case <-cc.done:
+			log.Infof("Shutting down...")
+		}
+
+		if atomic.SwapInt32(&cc.doneBool, 1) != 1 {
```
Comment (mheffner) [inline graceful/graceful.go]: What is `doneBool` for and where is it set?

**Comment by mheffner** [comment]
Oops looks like Ryan beat me to it :-)

**Pull request merged**